pkg/types/builder.go

```diff
@@ -429,10 +433,15 @@ func (r *resource) addObservationField(f *Field, field *types.Var) {
 	r.obsTags = append(r.obsTags, fmt.Sprintf(`json:"%s" tf:"%s"`, f.JSONTag, f.TFTag))
 }
 
-func (r *resource) addReferenceFields(g *Builder, paramName *types.TypeName, field *Field) {
+func (r *resource) addReferenceFields(g *Builder, paramName *types.TypeName, field *Field, isInit bool) {
 	refFields, refTags := g.generateReferenceFields(paramName, field)
-	r.paramTags = append(r.paramTags, refTags...)
-	r.paramFields = append(r.paramFields, refFields...)
+	if isInit {
+		r.initTags = append(r.initTags, refTags...)
+		r.initFields = append(r.initFields, refFields...)
+	} else {
+		r.paramTags = append(r.paramTags, refTags...)
+		r.paramFields = append(r.paramFields, refFields...)
+	}
```

Comment:
I think the reference fields will now always be added to the `spec.forProvider` and `spec.initProvider`, is this correct? If we don't have the branching here, what would cause an issue is the fact that this function is called once for the `forProvider` and once for `initProvider`, is this correct?

Reply:
Yes the both are correct. Now we always generate the reference fields for both. And if we do not have a branch here, this will cause an issue.